packages/algob/src/internal/deployer.ts

```diff
@@ -114,11 +117,30 @@ class DeployerBasicMode {
    * @returns {LogicSig} multi signed logic signature from assets/<file_name>.msig
    */
   async loadMultiSig (name: string, scParams: Object): Promise<LogicSig> {
-    const [tealFile, Msig] = await readMsigFromFile(name) as string; // Get tealFile name, Msig object from .mlsig
-    const lsig = await this.loadLsig(tealFile, scParams); // Load lsig from .teal (getting logic part from lsig)
-    const decodedMsig = await decodeMsigObj(Msig);
-    lsig.msig = {};
-    Object.assign(lsig.msig, decodedMsig);
+    const lsig = await getLsig(name, scParams, this.algoOp.algodClient); // get lsig from .teal (getting logic part from lsig)
+    const Msig = await readMsigFromFile(name); // Get decoded Msig object from .msig
+    Object.assign(lsig.msig = {}, Msig);
+    return lsig;
+  }
+
+  /**
+   * Description : loads multisigned logic signature from .msig file
+   * @param {string} name filename
+   * @returns {LogicSig} multi signed logic signature from assets/<file_name>.msig
+   */
+  async loadBinaryMultiSig (name: string): Promise<LogicSig> {
+    // get logic signature from file and decode it
+    const data = await readBinaryMultiSig(name);
+    const program = new Uint8Array(Buffer.from(data, 'base64'));
+    const logicSignature = decode(program) as LogicSig;
+
```

Comment:
can we use `return logicSignature` here and remove the succeeding lines ?

Reply:
We can't. We are decoding a simple object `{ l: [..], msig: {..} }`. In below lines we are creating an lsig and assigning this decoded object to lsig.

Reply:
Do you suggest something else?

Reply:
yes, why are we doing that ? we should load full multisig given by user and not distort it

Reply:
We are loading it totally for msig only, please check again. Assignment could be better.